### PR TITLE
add student profile update

### PR DESCRIPTION
**Pull request by palak199**
@garg3133 I have create the form for updating student profile. This is in reference to issue #73
 however I have some doubts:
* which fields are required
* which validations are needed 
Please review and let me know if there are other changes needed

### CODE REVIEW

**Comment by garg3133** [comment]
@palak199 I was expecting to have the structure of the update profile page to be same as the profile page, by just replacing the titles with labels and values with text fields. I am planning to change the Update Volunteer Profile page as well in the future.

For the questions you asked, all fields are required except the contact field. And validation needs to be done in contact field only (there should be exactly 10 digits).

**Comment by palak199** [comment]
> @palak199 I was expecting to have the structure of the update profile page to be same as the profile page, by just replacing the titles with labels and values with text fields. I am planning to change the Update Volunteer Profile page as well in the future.
> 
Okay, so I have done this on student page, please review.
> For the questions you asked, all fields are required except the contact field. And validation needs to be done in contact field only (there should be exactly 10 digits).

This is also done, you can check now.

**Comment by garg3133** [comment]
@palak199 I'm expecting something like this for the page structure (replacing title with labels and values with text fields, like I said before):
Screenshot from 2021-02-04 12-04-19

Also, you'll need to find a way to update the profile_pic too,

**Comment by palak199** [comment]
> @palak199 I'm expecting something like this for the page structure (replacing title with labels and values with text fields, like I said before):
> Screenshot from 2021-02-04 12-04-19

Now it looks as expected. 
> Also, you'll need to find a way to update the profile_pic too

The model doesn't have profile image attribute. :(

**Comment by garg3133** [comment]
Yeah, I think it would be better to have two inputs in one row. The form gets too long this way.

Apart from that:
* Form validations is not working (I'm able to enter a contact number with 9 digits)
* Village should be a drop-down menu with the village of the student selected (The method I had used for selecting gender is actually wrong. All the genders should have be displayed normally in the dropdown and the selection of student's gender can be easily done in JavaScript. The current approach requires us know beforehand all possible values of the dropdown.)
* There should be some margin below card so that its bottom border don't hide behind the footer.

For the layout, that's what I wanted :raised_hands:

**Comment by palak199** [comment]
> Yeah, I think it would be better to have two inputs in one row. The form gets too long this way.
> 
> Apart from that:
> 
> * Form validations is not working (I'm able to enter a contact number with 9 digits)

> * Village should be a drop-down menu with the village of the student selected (The method I had used for selecting gender is actually wrong. All the genders should have be displayed normally in the dropdown and the selection of student's gender can be easily done in JavaScript. The current approach requires us know beforehand all possible values of the dropdown.)
> * There should be some margin below card so that its bottom border don't hide behind the footer.
> 
> For the layout, that's what I wanted
>
screencapture-127-0-0-1-8000-students-profile-update-1-2021-02-05-11_42_39
Please have a look at the screenshot and I think all the improvements you made are done

File: templates/students/update_profile.html [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+                                    Please enter a valid class.
+                                </div>
+                        </div>
+                    </div>
+
+                        <div class="row justify-content-center">
+                            <div class="col-6 col-lg-6 mb-3 align-self-center">
+                                <label for="village">Village</label>
+                                <select name="village" class="custom-select d-block w-100" id="village" required>
+                                    {% for village, i in villages %}
```
Comment (garg3133) [inline templates/students/update_profile.html:85]: village_id, village_name can be used for better readability.

File: templates/students/update_profile.html [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+                                <label for="gender">Gender</label>
+                                <select name="gender" class="custom-select d-block w-100" id="gender" required>
+                                    <option value="M" {% if  profile.gender == "M" %} selected {% endif %} >Male</option>
+                                    <option value="F" {% if  profile.gender == "F" %} selected {% endif %} >Female</option>
+                                    <option value="O" {% if  profile.gender == "O" %} selected {% endif %} >Other</option>
+                                </select>
+                            </div>
+                            <div class="col-6 col-lg-6 mb-3">
+                                <label for="schoolClass">Class</label>
+                                <input name="school_class" type="number" class="form-control" id="school_class" value="{{ profile.school_class }}" min="0" max="12" required>
```
Comment (garg3133) [inline templates/students/update_profile.html:74]: input id should match with the for attribute of label.

Also, keep `min` as 1 and `max` as 15 (we keep 13 class for pass-out students and 14, 15 for Navodaya students).

File: templates/students/update_profile.html [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+                </div>
+        </div>
+
+                        <hr class="mb-3">
+                        <div class="row justify-content-center mb-3">
+                            <div class="col-md-6">
+                                <button name="submit" value="update-profile" class="btn btn-primary btn-lg btn-block" type="submit">Update</button>
+                            </div>
+                        </div>
+            </div>
```
Comment (garg3133) [inline templates/students/update_profile.html:122]: I think one `</div>` is missing here somewhere.

File: templates/students/update_profile.html [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+            </form>
+    </div>
+<script type="text/javascript">
```
Comment (garg3133) [inline templates/students/update_profile.html:126]: Put the script in `{% block scripts %}`

**Review by garg3133 — COMMENTED** [review]
The design looks cool now, exactly as I wanted. Thanks :smile: 

Here are a few changes which needs to be done. Also, format the HTML file properly (indentations and all).

**Comment by palak199** [comment]
I have made all the changes as stated above. Please review again @garg3133

**Comment by garg3133** [comment]
Changes seem good now! Although the formatting you did was horrifying (you removed all the empty lines). The formatting is done for better readability of code.

Merged the changes into master after correcting the formatting of HTML file.

Thanks a lot for your contribution 🚀

**Comment by palak199** [comment]
> Changes seem good now! Although the formatting you did was horrifying (you removed all the empty lines). The formatting is done for better readability of code.
> 
I used an online formatter for the file. :(
> Merged the changes into master after correcting the formatting of HTML file.
> 
> Thanks a lot for your contribution
>
Thanks for reviewing

**Pull request merged**